samples/net/dsa/src/dsa_lldp.c

```diff
@@ -18,24 +18,89 @@
 #include <errno.h>
 
 #include <zephyr/logging/log.h>
-#include "main.h"
 
 /* Loglevel of dsa_lldp function */
-LOG_MODULE_DECLARE(net_dsa_lldp_sample, CONFIG_NET_DSA_LOG_LEVEL);
+LOG_MODULE_DECLARE(net_dsa_sample, CONFIG_NET_DSA_LOG_LEVEL);
 
-#define LLDP_SYSTEM_NAME_SIZE 24
-#define LLDP_ETHER_TYPE 0x88CC
+#include "dsa_lldp.h"
+
+#define LLDP_SYSTEM_NAME_SIZE    24
+#define LLDP_ETHER_TYPE          0x88CC
 #define LLDP_INPUT_DATA_BUF_SIZE 512
-#define DSA_BUF_SIZ 128
-int dsa_lldp_send(struct net_if *iface, struct instance_data *pd,
-		  uint16_t lan, int src_port, int origin_port, int cmd,
-		  struct eth_addr *origin_addr)
+#define DSA_BUF_SIZ              128
+
+static const uint8_t eth_filter_l2_addr_base[][6] = {
+	/* MAC address of other device - for filtering testing */
+	{0x01, 0x80, 0xc2, 0x00, 0x00, 0x03}};
+
+enum net_verdict dsa_ll_addr_switch_cb(struct net_if *iface, struct net_pkt *pkt)
+{
+	struct net_eth_hdr *hdr = NET_ETH_HDR(pkt);
+	struct net_linkaddr lladst;
+
+	net_pkt_cursor_init(pkt);
+	lladst.len = sizeof(hdr->dst.addr);
+	lladst.addr = &hdr->dst.addr[0];
+
+	/*
+	 * Pass packet to lan1..3 when matching one from
+	 * check_ll_ether_addr table
+	 */
+	if (check_ll_ether_addr(lladst.addr, &eth_filter_l2_addr_base[0][0])) {
+		return 1;
+	}
+
+	return 0;
+}
+
+int start_slave_port_packet_socket(struct net_if *iface, struct instance_data *pd)
+{
+	struct sockaddr_ll dst;
+	int ret;
+
+	pd->sock = socket(AF_PACKET, SOCK_RAW, htons(ETH_P_ALL));
+	if (pd->sock < 0) {
+		LOG_ERR("Failed to create RAW socket : %d", errno);
+		return -errno;
+	}
+
+	dst.sll_ifindex = net_if_get_by_iface(iface);
+	dst.sll_family = AF_PACKET;
+
+	ret = bind(pd->sock, (const struct sockaddr *)&dst, sizeof(struct sockaddr_ll));
+	if (ret < 0) {
+		LOG_ERR("Failed to bind packet socket : %d", errno);
+		return -errno;
+	}
+
+	return 0;
+}
+
+void dsa_lldp(struct ud *user_data)
+{
+	uint8_t tbl_buf[8];
+
+	/*
+	 * Set static table to forward LLDP protocol packets
+	 * to master port.
+	 */
+	dsa_switch_set_mac_table_entry(user_data->lan[0], &eth_filter_l2_addr_base[0][0], BIT(4), 0,
+				       0);
+	dsa_switch_get_mac_table_entry(user_data->lan[0], tbl_buf, 0);
+
+	LOG_INF("DSA static MAC address table entry [%d]:", 0);
+	LOG_INF("0x%x 0x%x 0x%x 0x%x 0x%x 0x%x 0x%x 0x%x", tbl_buf[7], tbl_buf[6], tbl_buf[5],
+		tbl_buf[4], tbl_buf[3], tbl_buf[2], tbl_buf[1], tbl_buf[0]);
+}
+
+static int dsa_lldp_send(struct net_if *iface, struct instance_data *pd, uint16_t lan, int src_port,
+			 int origin_port, int cmd, struct eth_addr *origin_addr)
 {
 	int ret, len;
 	char buffer[DSA_BUF_SIZ], sys_name[LLDP_SYSTEM_NAME_SIZE];
 	struct sockaddr_ll dst;
 	struct ethernet_context *ctx = net_if_l2_data(iface);
-	struct net_eth_hdr *eth_hdr = (struct net_eth_hdr *) buffer;
+	struct net_eth_hdr *eth_hdr = (struct net_eth_hdr *)buffer;
```

Comment:
Could you place the cosmetic changes in this file to a separate commit?

Reply:
Sure. Thanks.